### PR TITLE
Make availability set optional (defaults to true) to allow creating more than 200 VMs

### PR DESCRIPTION
**Pull request by nakranimohit0**

### CODE REVIEW

File: wvd-templates/Create and provision WVD host pool/createUiDefinition.json [hunk 1]
```diff
@@ -91,6 +91,26 @@
 				"bladeTitle": "Configure number of VMs based on profile usage",
 				"bladeSubtitle": "Select a usage profile and number of users to determine the recommended number of VMs",
 				"elements": [
+					{
+						"name": "createAvailabilitySet",
+						"type": "Microsoft.Common.OptionsGroup",
+						"label": "Create an Availability Set",
+						"toolTip": "//todo decide one of: Using an Availability set limits you to a maximum of 200 virtual machines. Or An Availability sets can contain a maximum of only 200 virtual machines.",
```
Comment (nakranimohit0) [inline wvd-templates/Create and provision WVD host pool/createUiDefinition.json:98]: Need to update this

File: wvd-templates/Create and provision WVD host pool/mainTemplate.json [hunk 2]
```diff
@@ -256,6 +256,13 @@
             "metadata": {
                 "description": "(Required for Azure Marketplace.) Leave as is, unless you would like to not use a location that is different from the location of the resouce group."
             }
+        },
+        "createAvailabilitySet": {
+            "type": "bool",
+            "metadata": {
+                "description": "Set this parameter to true if you would like to create Availability set. Defaults to true. Please note that if 'enablePersistentDesktop' is true, this parameter will be ignored and no availability set wil be created. //todo add the tooltip text from UI here, update template & all the nested templates"
```
Comment (nakranimohit0) [inline wvd-templates/Create and provision WVD host pool/mainTemplate.json:263]: Need to update this

File: wvd-templates/Update existing WVD host pool/mainTemplate.json [hunk 3]
```diff
@@ -274,6 +274,13 @@
                 "description": "(Required for Azure Marketplace.) Leave as is, unless you would like to not use a location that is different from the location of the resouce group."
             }
         }
+        "createAvailabilitySet": {
+            "type": "bool",
+            "metadata": {
+                "description": "Set this parameter to true if you would like to create Availability set. Defaults to true. Please note that if 'enablePersistentDesktop' is true, this parameter will be ignored and no availability set wil be created. //todo add the tooltip text from UI here, update template & all the nested templates"
```
Comment (nakranimohit0) [inline wvd-templates/Update existing WVD host pool/mainTemplate.json:280]: Need to update this

File: wvd-templates/nestedtemplates/managedDisks-customimagevm.json [hunk 4]
```diff
@@ -104,19 +104,19 @@
                 "description": "Location for all resources to be created in."
             }
         },
-        "enablePersistentDesktop": {
+        "createAvailabilitySet": {
```
Comment (lintFan) [inline wvd-templates/nestedtemplates/managedDisks-customimagevm.json:107]: We need to wait on merging this until this change has been deployed to the WVD marketplace so we don't break re-deployments of the marketplace that points to master

**Comment by lintFan** [comment]
@stgeorgi please check for terraform compatibility

**Comment by stgeorgi** [comment]
are there changes to the DSC's that is the only thing I use from the main repo

**Comment by lintFan** [comment]
@stgeorgi  There are no DSC changes so it should be fine then.

**Comment by stgeorgi** [comment]
I can still test are all the changes live in the master github for me to go ahead and test

**Comment by lintFan** [comment]
@stgeorgi  No, we are waiting for the marketplace to be deployed before this is merged to master for compatibility reasons. But you can test using this branch user_monakran_optional_avail_set if you would like.

**Pull request merged**